### DISCUSSION TITLE
[Feature]: Able to set custom docker compose up command for compose buildpacks

### DISCUSSION THREAD

**Discussion by andrasbacsai** [coollabsio/coolify#3133] [Feature Requests]
### Description

👋 

### Minimal Reproduction (if possible, example repository)

🌮 

### Version

latest v4 beta

**Comment by jnreis** [comment]
> **Note:** After a publishing this comment I realized that this already exists when creating a new Application. It is possible to use a docker-compose file in the repository to create a new app.

Hi @andrasbacsai this feature would help me very much.

Instead of using one of [the predefined Services](https://github.com/coollabsio/coolify/tree/main/templates), it would be great be able to write my own using the Coolify UI.

The **Select a Service Type** page could have an option called **Custom Service**.

In the Configurations tab after the General section, add a new section called Compose with a simple textarea (or a fancier yaml editor with code highlight - it doesn't matter). Also with the ability to use any environment variable defined in the Secrets tab.

This feature would give me immense flexibility. Allowing me to try projects not yet available as a Coolify service.

Moreover, a button could also be added to "Share" the Service with the Coolify community. This would open a modal (or review a section) with all the options necessary to add a new service to the Coolify main branch. The modal would include things like the service name, yaml code (pre-populated from the open service but editable), env textarea for the essential environment variables (env variable names may also be pre-populated), service documentation link, service slogan, tags, and any other optional information that might be relevant, such as the Github username of the person that is sending the contribution (for credit purpose).

The service could then be send to a "community-templates" branch or tagged as community until it is proved that is stable to go the main branch or have the "community" tag removed.

I do believe this would quickly give Coolify an enormous service gallery, and would make Coolify a much more compelling Heroku alternative.
